qutebrowser/browser/commands.py

```diff
@@ -832,6 +833,51 @@ def paste(self, sel=False, tab=False, bg=False, window=False):
             self._open(url, tab, bg, window)
 
     @cmdutils.register(instance='command-dispatcher', scope='window',
+                       completion=[usertypes.Completion.tab])
+    def buffer(self, index):
+        """Select tab by index or url/title best match, focus window if
+        necessary.
+
+        Args:
+            index: The [win_id/]index of the tab to focus. Or a substring
+                   in which case the closest match will be focused.
+        """
+        try:
+            for part in index.split('/'):
+                int(part)
+        except ValueError:
+            model = instances.get(usertypes.Completion.tab)
+            sf = sortfilter.CompletionFilterModel(source=model)
+            sf.set_pattern(index)
+            if sf.count() > 0:
+                index = sf.data(sf.first_item())
+            else:
+                raise cmdexc.CommandError(
+                    "No matching tab for: {}".format(index))
+
+        if len(index.split('/')) == 2:
+            win_id, idx = map(int, index.split('/'))
```

Comment:
I don't like `map` :wink:

I think this would be better written as:

```python
index_parts = index.split('/')

if len(index_parts) == 2:
    win_id = int(index_parts[0])
    idx = int(index_parts[1])
```

(then you can also re-use `index_parts` for the `elif` below)